spec.html

```diff
@@ -430,12 +430,12 @@ <h1>CreateSegmentDataObject ( _segmenter_, _string_, _startIndex_, _endIndex_ )<
         1. Assert: _endIndex_ &le; _len_.
         1. Assert: _startIndex_ &lt; _endIndex_.
         1. Let _result_ be ! ObjectCreate(%ObjectPrototype%).
-        1. Let _segment_ be the String value containing consecutive code units from _string_ beginning with the code unit at index _startIndex_ and ending with the code unit at index _endIndex_ - 1.
+        1. Let _segment_ be the String value equal to the substring of _string_ consisting of the code units at indices _startIndex_ (inclusive) through _endIndex_ (exclusive).
         1. Perform ! CreateDataPropertyOrThrow(_result_, *"segment"*, _segment_).
         1. Perform ! CreateDataPropertyOrThrow(_result_, *"index"*, _startIndex_).
         1. Let _granularity_ be _segmenter_.[[SegmenterGranularity]].
         1. If _granularity_ is `"word"`, then
-          1. Let _isWordLike_ be a Boolean value indicating whether the word segment _segment_ in _string_ is "word-like" according to locale _segmenter_.[[Locale]].
+          1. Let _isWordLike_ be a Boolean value indicating whether _segment_ is "word-like" according to locale _segmenter_.[[Locale]].
```

Comment:
> I don't like this change; it precludes the possibility of using context to decide whether or not a segment is word-like.

ok. that make sense, how about just change "word segment _segment_" to "_segment_" so it read as 
"whether the  _segment_ in _string_ is "word-like""
instead of
"whether the word segment _segment_ in _string_ is "word-like""

The concept of "word segment ... is "word-like"" is just circulation definition. if it is a word segment, then it must be "word-like". If it is "word-like" it must be a "word" segment, right?

Reply:
That works, but also would make more sense in a PR that wasn't attempting to refactor FindBoundary.